src/Simplex/Messaging/Agent.hs

```diff
@@ -1265,6 +1290,25 @@ abortConnectionSwitch' c connId =
         _ -> throwError $ CMD PROHIBITED
       _ -> throwError $ CMD PROHIBITED
 
+synchronizeConnectionRatchet' :: AgentMonad m => AgentClient -> ConnId -> Bool -> m ConnectionStats
+synchronizeConnectionRatchet' c connId force = withConnLock c connId "synchronizeConnectionRatchet" $ do
+  withStore c (`getConn` connId) >>= \case
+    SomeConn _ (DuplexConnection cData@ConnData {ratchetSyncState} rqs sqs)
+      | ratchetSyncState `elem` ([RSAllowed, RSRequired] :: [RatchetSyncState]) || force -> do
+        -- check queues are not switching?
```

Comment:
and probably also check ratchet is not synching while queues are switching?

Reply:
there is

Reply:
I'm not sure about checking queues are not switching when trying to synchronize ratchet though, because it's pointless to wait for queue switch messages if they won't decrypt